### PR TITLE
wasapi: Fix compilation for VS2010.

### PR DESCRIPTION
**Pull request by alienpenguin**
Fix WASAPI compilation on VS2010
- use C89 syntax for variable declaration
- use "_inline" keyword instead of "inline" when compiling with VS2010 and earlier

### CODE REVIEW

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 1]
```diff
@@ -2078,6 +2086,7 @@ static PaDeviceInfo *AllocateDeviceListMemory(PaWasapiHostApiRepresentation *paW
 
     if (paWasapi->deviceCount != 0)
     {
+		UINT32 i;
```
Comment (MichalPetryka) [inline src/hostapi/wasapi/pa_win_wasapi.c:2089]: Bad indent

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 2]
```diff
@@ -6087,10 +6096,11 @@ static UINT32 ConfigureLoopSleepTimeAndScheduler(PaWasapiStream *stream, ThreadI
 }
 
 // ------------------------------------------------------------------------------------------
-static inline INT32 GetNextSleepTime(SystemTimer *timer, ThreadIdleScheduler *scheduler, LONGLONG startTime,
+static PA_INLINE INT32 GetNextSleepTime(SystemTimer *timer, ThreadIdleScheduler *scheduler, LONGLONG startTime,
     UINT32 sleepTime)
 {
     INT32 nextSleepTime;
+	UINT32 procTime;
```
Comment (MichalPetryka) [inline src/hostapi/wasapi/pa_win_wasapi.c:6103]: Same here

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 3]
```diff
@@ -306,6 +306,14 @@ typedef struct _pa_AudioClientProperties {
     #define PA_THREAD_ID DWORD
 #endif
 
+
+//Visual Studio 2010 does not like the inline keyword
+#if _MSC_VER <= 1600
+    #define PA_INLINE _inline
+#else
+    #define PA_INLINE inline
+#endif
```
Comment (dmitrykos) [inline src/hostapi/wasapi/pa_win_wasapi.c:315]: I propose to modify this whole block to:

```c
#if (_MSC_VER <= 1600)
    #define inline _inline
#endif
```

and move it inside the compatibility block `#if defined(_MSC_VER) && (_MSC_VER >= 1400)` right after the line https://github.com/PortAudio/portaudio/blob/8ccd53fe7ced184c67fcd5d22939a2cf383b4597/src/hostapi/wasapi/pa_win_wasapi.c#L89

And as a result please undo all changes where inline was replaced with PA_INLINE.

It will be much cleaner compatibility fix without an intrusion to a code which is compliant with modern compilers.

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 4]
```diff
@@ -811,7 +819,7 @@ static void ThreadIdleScheduler_Setup(ThreadIdleScheduler *sched, UINT32 resolut
 }
 
 //! Iterate and check if can sleep.
-static inline UINT32 ThreadIdleScheduler_NextSleep(ThreadIdleScheduler *sched)
+static PA_INLINE UINT32 ThreadIdleScheduler_NextSleep(ThreadIdleScheduler *sched)
```
Comment (dmitrykos) [inline src/hostapi/wasapi/pa_win_wasapi.c:822]: Undo this change.

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 5]
```diff
@@ -886,7 +894,7 @@ static void SystemTimer_InitializeTimeGetter()
 }
 
 //! Get high-resolution time in milliseconds (using QPC by default).
-static inline LONGLONG SystemTimer_GetTime(SystemTimer *timer)
+static PA_INLINE LONGLONG SystemTimer_GetTime(SystemTimer *timer)
```
Comment (dmitrykos) [inline src/hostapi/wasapi/pa_win_wasapi.c:897]: Undo this change.

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 6]
```diff
@@ -6104,7 +6114,7 @@ static inline INT32 GetNextSleepTime(SystemTimer *timer, ThreadIdleScheduler *sc
     //
     // [9],{2},[8],{1},[9],{1},[9],{3},[7],{2},[8],{3},[7],{2},[8],{2},[8],{3},[7],{2},[8],...
     //
-    INT32 procTime = (INT32)(SystemTimer_GetTime(timer) - startTime);
```
Comment (dmitrykos) [inline src/hostapi/wasapi/pa_win_wasapi.c:6107]: Please change of the type of `procTime` back to`INT32`.

**Review by dmitrykos — CHANGES_REQUESTED** [review]

**Comment by dmitrykos** [comment]
@alienpenguin please double check indentation of all the lines you changed and squash all your changes (initial and subsequent) into a single commit.

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 2 above)
Comment (dmitrykos) [reply src/hostapi/wasapi/pa_win_wasapi.c:6103]: `UINT32 procTime` must be changed back to `INT32 procTime`.

Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c]: it was a typo, i will change it back to INT32

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 1 above)
Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c]: my bad, VS IDE was not configured to use spaces instead of tabs

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 3 above)
Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c]: agreed and done

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 4 above)
Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c]: done

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 5 above)
Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c]: done

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 6 above)
Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c:6107]: done

File: src/hostapi/wasapi/pa_win_wasapi.c [hunk 7]
```diff
@@ -87,6 +87,10 @@
     #include <mmdeviceapi.h>
     #include <devicetopology.h>    // Used to get IKsJackDescription interface
     #undef INITGUID
+//Visual Studio 2010 does not like the inline keyword
```
Comment (dmitrykos) [inline src/hostapi/wasapi/pa_win_wasapi.c:90]: Would you please add space after the // and modify `does not like the inline keyword` to `does not support the inline keyword`. After this change all will be ok and will be able to merge it.

**Review by dmitrykos — CHANGES_REQUESTED** [review]

File: src/hostapi/wasapi/pa_win_wasapi.c (hunk 7 above)
Comment (alienpenguin) [reply src/hostapi/wasapi/pa_win_wasapi.c]: done as requested

**Review by dmitrykos — APPROVED** [review]

**Review by dmitrykos — APPROVED** [review]

**Comment by dmitrykos** [comment]
@alienpenguin, thank you!

**Comment by oifii** [comment]
For successfully compiling pa_win_wasapi.c with VS2013, I also had to have inline redefined as _inline. I took the source code from portaudio github master branch cloned back on 2021july27.

My patch to pa_win_wasapi.c is:

//2022oct19, spi, begin
// Visual Studio 2010 does not support the inline keyword
//#if (_MSC_VER <= 1600)
// Visual Studio 2013's C compiler does not support the inline keyword when preceeded of static
//note: VS2013's C compiler is C89/C90 plus some C99 features.
#if (_MSC_VER <= 1800)
//2022oct19, spi, end
    #define inline _inline
#endif

**Comment by oifii** [comment]
For successfully compiling pa_win_util.c with VS2010, a source code snapshot from portaudio github master branch cloned back on 2021july27, I had to slightly modify function PaWinUtil_SetLastSystemErrorInfo()'s implementation. It only required the swapping of two lines of code, i.e. declaring line "char msg[1024];" before calling function FormatMessageW();
